students/chanjoo/westargram/users/views.py

```diff
@@ -1,3 +1,39 @@
-from django.shortcuts import render
+import json
+from re import M
 
-# Create your views here.
+from django.http            import JsonResponse
+from django.views           import View
+from django.core.exceptions import ValidationError
+
+from users.models      import User
+from users.validations import validate_password, validate_phone
+
+class SignUpView(View):
+    def post(self, request):
+        try:
+            user_info = json.loads(request.body)
+
+            phone_number = user_info['phone_number']
+            name         = user_info['name']
+            username     = user_info['username']
+            password     = user_info['password']
+
+            validate_phone(phone_number)
+            validate_password(password)
+
+            if User.objects.filter(phone_number=phone_number).exists():
+                raise ValidationError('이미 가입된 핸드폰 번호입니다.')
```

Comment:
```suggestion
                raise ValidationError('invalid phon_number')
```
메시지는 영어로 간결하게 해주세요. 저희가 한국인하고만 협업하는 것이 아닙니다. 따라서 공통어인 영어를 사용해 주세요. 그리고 이 메시지는 사용자가 아닌 프론트엔드가 보는 메시지이기 때문에 자세하게 설명하지 않으셔도 됩니다.(메시지는 예시일 뿐이니 적절한 메시지를 넣어주시면 됩니다.)

Reply:
오.. 딱 궁금했던 부분인데 명쾌하게 설명해주셔서 감사합니다 ! 수정하겠습니다